StrangeIoC/scripts/strange/extensions/mediation/impl/AbstractMediationBinder.cs

```diff
@@ -0,0 +1,278 @@
+			{
+				switch(evt)
+				{
+					case MediationEvent.AWAKE:
+						InjectViewAndChildren(view);
+						MapView (view, binding);
+						break;
+					case MediationEvent.DESTROYED:
+						UnmapView (view, binding);
+						break;
+					default:
+						break;
+				}
+			}
+			else if (evt == MediationEvent.AWAKE)
+			{
+				//Even if not mapped, Views (and their children) have potential to be injected
+				InjectViewAndChildren(view);
+			}
+		}
+
+		/// Add a Mediator to a View. If the mediator is a "true" Mediator (i.e., it
+		/// implements IMediator), perform PreRegister and OnRegister.
+		protected virtual void ApplyMediationToView(IMediationBinding binding, IView view, Type mediatorType)
+		{
+			bool isTrueMediator = IsTrueMediator(mediatorType);
+			if (!isTrueMediator || !HasMediator(view, mediatorType))
+			{
+				Type viewType = view.GetType();
+				object mediator = CreateMediator(view, mediatorType);
+
+				if (mediator == null)
+					ThrowNullMediatorError (viewType, mediatorType);
+				if (isTrueMediator)
+					((IMediator)mediator).PreRegister();
+
+				Type typeToInject = (binding.abstraction == null || binding.abstraction.Equals(BindingConst.NULLOID)) ? viewType : binding.abstraction as Type;
+				injectionBinder.Bind(typeToInject).ToValue(view).ToInject(false);
+				injectionBinder.injector.Inject(mediator);
+				injectionBinder.Unbind(typeToInject);
+				if (isTrueMediator)
+				{
+					((IMediator)mediator).OnRegister();
+				}
+			}
+		}
+
+		/// Add Mediators to Views. We make this virtual to allow for different concrete
+		/// behaviors for different View/Mediation Types (e.g., MonoBehaviours require 
+		/// different handling than EditorWindows)
+		protected virtual void InjectViewAndChildren(IView view)
+		{
+			IView[] views = GetViews(view);
+			int aa = views.Length;
+			for (int a = aa - 1; a > -1; a--)
+			{
+				IView iView = views[a] as IView;
+				if (iView != null)
+				{
+					if (iView.autoRegisterWithContext && iView.registeredWithContext)
+					{
+						continue;
+					}
+					iView.registeredWithContext = true;
+					if (iView.Equals(view) == false)
+						Trigger(MediationEvent.AWAKE, iView);
+				}
+			}
+			injectionBinder.injector.Inject(view, false);
+		}
+
+		protected virtual bool IsTrueMediator(Type mediatorType)
+		{
+			return typeof(IMediator).IsAssignableFrom(mediatorType);
+		}
+
+		override protected IBinding performKeyValueBindings(List<object> keyList, List<object> valueList)
+		{
+			IBinding binding = null;
+
+			// Bind in order
+			foreach (object key in keyList)
+			{
+				Type keyType = Type.GetType (key as string);
+				if (keyType == null)
+				{
+					throw new BinderException ("A runtime Mediation Binding has resolved to null. Did you forget to register its fully-qualified name?\n View:" + key, BinderExceptionType.RUNTIME_NULL_VALUE);
+				}
+				if (binding == null)
+				{
+					binding = Bind (keyType);
+				}
+				else
+				{
+					binding = binding.Bind (keyType);
```

Comment:
Nah, it's a mistake. The default keyConstraint is for a Binding is BindingConstraintType.ONE, and MediationBinding doesn't override that. So this condition can never arise.